### PR TITLE
Thai/include deleted activities

### PR DESCRIPTION
**Pull request by athai**
Add 'include_deleted' parameter to the activities endpoint.

### CODE REVIEW

**Comment by athai** [comment]
Question: If a user tries to specify the deleted activity by 'activities/:slug?include_deleted=true', what error should be returned? I'm going with Object Not Found for now, but is there a better alternative?

**Comment by ElijahCaine** [comment]
@athai Object Not Found is the correct error afaik.

**Comment by LyonesGamer** [comment]
IMO Object Not Found is the best here.

**Comment by athai** [comment]
@ElijahCaine @LyonesGamer Cool, thanks. Also, how do my tests look?

File: tests/activities.js [hunk 1]
```diff
@@ -347,6 +347,158 @@ module.exports = function(expect, request, baseUrl) {
...
+        ];
+
+        expect(getErr).to.equal(null);
+        expect(getRes.statusCode).to.equal(200);
+        expect(jsonBody).to.deep.equal(expectedResults);
+        done();
+      });
+    });
+
+    it('returns an error if user specifies with /activities/:slug endpoint',
```
Comment (ElijahCaine) [inline tests/activities.js]: This should describe that the slug does not exist. Sending /activities/ a :slug is valid.

**Comment by ElijahCaine** [comment]
:thumbsup: looks good to me.

**Comment by LyonesGamer** [comment]
+1

File: src/activities.js [hunk 2]
```diff
@@ -8,7 +8,17 @@ module.exports = function(app) {
...
   app.get(app.get('version') + '/activities', function(req, res) {
     const knex = app.get('knex');
-    knex('activities').then(function(activities) {
+
+    // Only query for activities that have not been soft-deleted
+    let activitiesQ = knex('activities').whereNull('deleted_at');
+
+    // If the include_deleted param is true, query for all activities
+    // regardless of 'deleted' status
+    if (req.query.include_deleted) {
```
Comment (ElijahCaine) [inline src/activities.js]: Is there a reason this is two separate statements and not an if/else? Querying the database twice if the parameter is passed seems like a needless performance hit.

Comment (athai) [inline src/activities.js]: How would you encase this in an if/else?

Comment (ElijahCaine) [inline src/activities.js]: I was thinking something like this: 
```
if (req.query.include_deleted) {
  let activitiesQ = knex('activities');
} else {
  let activitiesQ = knex('activities').whereNull('deleted_at');
}
```

Comment (LyonesGamer) [inline src/activities.js]: Or to revise that slightly,

    let activitiesQ = knex('activities');
    if (req.query.include_deleted) {
      activitiesQ = activitiesQ.whereNull('deleted_at');
    }

**Comment by Kennric** [comment]
Tests pass, and the code looks good to me. Get a second opinion. +1

**Comment by ElijahCaine** [comment]
I agree with @Kennric on this. The code looks great. +1

**Pull request merged**